[blog] Add a post for the Tree View migration (#38407)
mui/material-ui: docs/pages/blog/lab-tree-view-to-mui-x.md

---
title: The Tree View is moving to MUI X
description: Migrate to the new package to start building with our powerful Tree View, now part of MUI X. Previously released MIT components will stay MIT.
date: 2023-08-11T00:00:00.000Z
authors: ['flaviendelangle']
tags: ['MUI X', 'News']

> card: true

Reviewer: ```suggestion
card: false
```

unless you will add card later?

Author: Fixed :+1:

Reviewer 2: It's still broken https://robolly.com/open-graph-preview/?url=https://mui.com/blog/lab-tree-view-to-mui-x/

<img width="462" alt="Screenshot 2023-09-06 at 20 56 52" src="https://github.com/mui/material-ui/assets/3165635/205cfdc1-792f-412d-8cff-af1a3e7a6c6a">

It's fixed in #38834.

---

After more than 18 months in the lab, the date and time picker components have found a new home as part of MUI X.
This means we'll be dedicating even more time and effort to these complex components, to better meet the needs of both you and your users.

## TL;DR

- The Tree View is one step closer to a stable release.
- **No surprise licenses changes**. We are staying true to [our promises](https://mui-org.notion.site/Stewardship-542a2226043d4f4a96dfb429d16cf5bd). What's MIT stays MIT. Therefore, all the existing features and a lot of the future features of the Tree View will remain MIT and free to use.
- Follow the [migration steps](/x/migration/migration-tree-view-lab/)

> ## What is the tree view?

Author: I have no idea what to write here :laughing:

The Tree View is a component to represent hierarchical data presented as nodes in a tree-like format.

The component allows to select one or multiple nodes.

## What is MUI X?

[MUI X](/x/) is a collection of advanced components built for complex use cases.

As opposed to the MUI Core library, which leans on the open-source community for support, MUI X components require several full-time developers dedicated to engineering and ongoing maintenance.

> MUI X components are available under two licenses:

Reviewer: @joserodolfofreitas at the moment, there's only two licenses, correct? I don't think we need to talk about "planned" positions.

> MUI X components are available under two licenses:

Reviewer 3: My point is that this section of the text talks about MUI X and mentions MIT and Pro, but doesn't mention Premium. 

```suggestion
MUI X components are available under three licenses:
```

## TL;DR

- The Tree View is one step closer to a stable release.
- **No surprise licenses changes**. We are staying true to [our promises](https://mui-org.notion.site/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).
  What's MIT stays MIT.
  Therefore, all the existing features and future features of the Tree View will remain MIT and free to use.
- Follow the [migration steps](/x/migration/migration-tree-view-lab/)

## What is the Tree View?

The Tree View is a component to represent hierarchical data presented as nodes in a tree-like format.

The component allows to select one or multiple nodes.

<video preload="metadata" style="margin-bottom: 16px;" autoplay muted loop>
  <source src="/static/blog/lab-tree-view-to-mui-x/treeview.mov" type="video/mp4">
</video>

## What is MUI X?

[MUI X](/x/) is a collection of advanced components built for complex use cases.

As opposed to the MUI Core library, which leans on the open-source community for support, MUI X components require several full-time developers dedicated to engineering and ongoing maintenance.

MUI X components are available under two licenses:

> - MUI X is MIT licensed, so it's free to use.

Reviewer 3: ```suggestion
- **MIT license**, which is free to use and includes all the current (and more to come) Treeview features.
```

> - MUI X Pro includes the most advanced features and is commercially licensed.

Reviewer 3: ```suggestion
- **MUI X Pro**, a commercial license that offers features for handling large datasets and will include the most advanced features for Treeview, like virtualization.
- **MUI X Premium**, which is also commercial and includes, in addition to the Pro features, tooling for data analysis.
```

> ## Why did the Tree View move to MUI X?

Reviewer 3: I think one of the reasons we could raise is that we want to go beyond what's offered in the MIT space.

I would perhaps argue like this:

"Once a component is ready to leave the lab, it can either go to MUI X or MUI Core libraries. 
Given the nature of the Treeview components, it was a clear choice to group it with the other advanced components. 
Besides, we plan to improve it beyond what's offered in the MIT space with extra fine-tuned features for enterprise and data-rich applications."

> When the Tree View component was ready to leave the lab, we had a choice to make: live in the MUI Core library or become a part of MUI X?

Reviewer 3: ```suggestion
Once a component is ready to leave the lab, it can either go to MUI X or MUI Core libraries.
```

> This component would greatly benefit from advanced features such as virtualization of drag and drop.

Reviewer 3: ```suggestion
Given the complex nature of the Treeview, it was a clear choice to group it with the other advanced components in MUI X.

Moreover, new advanced features such as virtualization and drag and drop are already on the roadmap.
```

> Given the complex nature of these features, we decided that it would benefit most from the dedicated attention that they would receive from the X team.

Reviewer 3: I don't think we should argue that they will receive "better" attention on X.

```suggestion
```

> The Tree View components is now available in the `@mui/x-tree-view` package which has a [MIT license](https://unpkg.com/browse/@mui/x-tree-view/LICENSE).

Reviewer 3: ```suggestion
The Tree View components are now available in the `@mui/x-tree-view` package which is [MIT licensed](https://unpkg.com/browse/@mui/x-tree-view/LICENSE) and includes all the current (free forever) features.
```

## How do I migrate?

Follow the [migration steps](/x/migration/migration-tree-view-lab/) by updating the package name and change from a default export to a named export:

```diff
-import TreeView from '@mui/lab/TreeView';
+import { TreeView } from '@mui/x-tree-view/TreeView';

-import { TreeView, TreeItem } from '@mui/lab';
+import { TreeView, TreeItem } from '@mui/x-tree-view';
```

We have prepared a codemod to help you migrate your codebase from `@mui/lab` to `@mui/x-tree-view`:

```bash
npx @mui/codemod v5.0.0/tree-view-moved-to-x <path>
```

## Where is the Tree View documentation?

You can find the documentation for the Tree View component in the [MUI X docs](/x/react-tree-view/).

## What's next for the Tree View?

For now, `@mui/x-tree-view` is in alpha.

> Our next goal is to work on the stability and API consistency of its components to prepare a stable release.

Reviewer 3: Should we set an expectation (and incentive)?

```suggestion
Our next goal is to work on the stability and API consistency of its components to prepare a stable release on the next few months.
```
